### ISSUE TITLE
Unable to alter/remove the `X-Powered-By` header? SECURITY problem? 

### ISSUE THREAD

**Issue by BrokenSourceCode** [ratchetphp/Ratchet#960]
Hi,

The `X-Powered-By` header sent by Ratchet describes the technologies used by the server. Therefore this info exposes the server to malicious attackers. Using the information in this header, attackers can find vulnerabilities easier.

**Comment by szado** [comment]
I agree that this is a security issue. Overall, trying to change something small with the decorators in Ratchet is difficult. The Ratchet methods are huge, and often you need to copy-paste and overwrite a large piece of library code to create your own decorator version.

**Comment by BrokenSourceCode** [comment]
> I agree that this is a security issue. Overall, trying to change something small with the decorators in Ratchet is difficult. The Ratchet methods are huge, and often you need to copy-paste and overwrite a large piece of library code to create your own decorator version.

Unfortunately, copy-paste and overwrite a large piece of library code is not a relevant solution for a security issue. It would be nice if we could easily change this very annoying header. Maybe with an argument given in the constructors?

**Comment by clue** [comment]
@Elysiome Thanks for reporting, I agree that `X-Powered-By` could be problematic and could potentially be used during reconnaissance to gain more information to prepare attacks and as such I agree this should be addressed.

I see you've already filed #992 and https://github.com/ratchetphp/RFC6455/pull/69 to address this, so let's continue discussion from there. This project is actively supported, but please keep in mind this is an open-source project that is mostly supported by its maintainers in their free time as good as humanly possible. Please reach out if you want to support this project.

In the meantime, you can always avoid this header by configuring your reverse proxy to remove this header. For nginx this should be as easy as:

```
proxy_hide_header X-Powered-By;
```

That said, I do not currently see that this HTTP header alone constitutes a security issue on its own. If you believe you have identified a security issue, I would like to ask you to follow responsible disclosure principles. See https://github.com/ratchetphp/Ratchet/security/policy for details.
